File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        ans = []
        # Determine the smallest array 
        min_length = min(len(word1), len(word2))
        # Iterate through this length adding element from word1 and then word2 into ans
        for i in range(min_length):
            ans.append(word1[i])
            ans.append(word2[i])
        # If word1 length > word2 length, iterate from minlength to word1 length and add to answer
        if len(word1) > len(word2):
            for i in range(min_length, len(word1)):
                ans.append(word1[i])
        # If word2 length > word1 length, iterate from minlength to word2 length and add to answer
        if len(word2) > len(word1):
            for i in range(min_length, len(word2)):
                ans.append(word2[i])
        # Return ans
        return "".join(ans)

File: 1894-merge-strings-alternately/merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let mergedWord = '';
    for(let i = 0; i < word1.length; i++) {
        mergedWord += word1[i];
        if(word2[i] !== undefined) {
            mergedWord += word2[i];
        }
    }

    if(word2[word1.length] !== undefined) {
        mergedWord += word2.slice(word1.length);
    }

    return mergedWord;
};